src/Plugin/Block/MemberCountBlock.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Drupal\og\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\og\MembershipManagerInterface;
+use Drupal\og\OgContextInterface;
+use Drupal\og\OgMembershipInterface;
+use Drupal\og\OgRoleInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a block that shows the number of members in the current group.
+ *
+ * This block is mainly intended to demonstrate the group membership list cache
+ * tag but can also be used to show the number of members on group pages. The
+ * way the text is displayed can be changed by overriding the Twig template.
+ *
+ * @Block(
+ *   id = "og_member_count",
+ *   admin_label = @Translation("Group member count")
+ * )
+ */
+class MemberCountBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The OG context provider.
+   *
+   * @var \Drupal\og\OgContextInterface
+   */
+  protected $ogContext;
+
+  /**
+   * The membership manager.
+   *
+   * @var \Drupal\og\MembershipManagerInterface
+   */
+  protected $membershipManager;
+
+  /**
+   * Constructs a MemberCountBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\og\OgContextInterface $og_context
+   *   The OG context provider.
+   * @param \Drupal\og\MembershipManagerInterface $membership_manager
+   *   The membership manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, OgContextInterface $og_context, MembershipManagerInterface $membership_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->ogContext = $og_context;
+    $this->membershipManager = $membership_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('og.context'),
+      $container->get('og.membership_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
```

Comment:
We always have to account the parent return value, even in this case that returns an empty array. This is because the parent might return something in the future or at some point we switch to other base class:

```
return [
  ...
] + parent:: defaultConfiguration();
```